share/p2p/shrexeds/client.go

```diff
@@ -131,3 +131,28 @@ func (c *Client) doRequest(
 		return nil, p2p.ErrInvalidResponse
 	}
 }
+
+func (c *Client) setStreamDeadlines(ctx context.Context, stream network.Stream) {
+	// set read/write deadline to use context deadline if it exists
+	if dl, ok := ctx.Deadline(); ok {
+		if err := stream.SetDeadline(dl); err != nil {
+			log.Debugw("client: setting deadline: %s", "err", err)
+		}
+	}
+
+	// if deadline not set, client read deadline defaults to server write deadline
+	if c.params.ServerWriteTimeout != 0 {
+		err := stream.SetReadDeadline(time.Now().Add(c.params.ServerWriteTimeout))
+		if err != nil {
+			log.Debugw("client: setting read deadline", "err", err)
+		}
+	}
+
+	// if deadline not set, client write deadline defaults to server read deadline
+	if c.params.ServerReadTimeout != 0 {
```

Comment:
Server timeouts on the client?

Reply:
This is how its done in shrexnd. It makes sense, maybe we could add some constant for the actual networking though

Reply:
This is setting same timeout for the other side of pipe. We could do separate client side timeouts later if we would need, but as for default settings those seems reasonable to mirror server side.